packages/smooth_app/lib/pages/product/product_image_viewer.dart

```diff
@@ -0,0 +1,141 @@
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:http/http.dart' as http;
+import 'package:path_provider/path_provider.dart';
+import 'package:photo_view/photo_view.dart';
+import 'package:smooth_app/data_models/product_image_data.dart';
+import 'package:smooth_app/generic_lib/design_constants.dart';
+import 'package:smooth_app/generic_lib/loading_dialog.dart';
+import 'package:smooth_app/generic_lib/widgets/smooth_back_button.dart';
+import 'package:smooth_app/pages/product/confirm_and_upload_picture.dart';
+import 'package:smooth_app/widgets/smooth_scaffold.dart';
+
+/// Displays a full-screen image with an edit floating button
+class ProductImageViewer extends StatefulWidget {
+  const ProductImageViewer({
+    required this.barcode,
+    required this.imageData,
+  });
+
+  final String barcode;
+  final ProductImageData imageData;
+
+  @override
+  State<ProductImageViewer> createState() => _ProductImageViewerState();
+}
+
+class _ProductImageViewerState extends State<ProductImageViewer> {
+  late final ProductImageData imageData;
+
+  /// When the image is edited, this is the new image
+  ImageProvider? imageProvider;
+  bool _isEdited = false;
+
+  @override
+  void initState() {
+    imageData = widget.imageData;
+
+    if (imageData.imageUrl != null) {
```

Comment:
It cannot be `final` as if the user changes it we change the `imageProvider` to be a `FileImage`

https://github.com/VaiTon/smooth-app/blob/9424fe8e8f5af5f2f97d1ad93abc68717691f7e8/packages/smooth_app/lib/pages/product/product_image_viewer.dart#L123

Also for the null check, I really don't know if it can be null. We're checking for the nullity of `imageUrl` in a lot of places so I would gladly postpone it to another PR